File: 3668.restore-finishing-order.py
#
# @lc app=leetcode id=3668 lang=python
#
# [3668] Restore Finishing Order
#

# @lc code=start
class Solution(object):
    def recoverOrder(self, order, friends):
        """
        :type order: List[int]
        :type friends: List[int]
        :rtype: List[int]
        """
        friend = set()
        for i in friends:
            friend.add(i)
        result = []
        for i in order:
            if i in friend:
                result.append(i)
        return result
        
# @lc code=end

File: 3668.restore-finishing-order.java
/*
 * @lc app=leetcode id=3668 lang=java
 *
 * [3668] Restore Finishing Order
 */

// @lc code=start

import java.util.ArrayList;
import java.util.HashSet;
import java.util.List;

class Solution {
    public int[] recoverOrder(int[] order, int[] friends) {
        HashSet<Integer> friend = new HashSet<>();
        for (int i : friends) {
            friend.add(i);
        }
        List<Integer> result = new ArrayList<>();
        for (int i : order) {
            if (friend.contains(i)) {
                result.add(i);
            }
        }
        return result.stream().mapToInt(Integer::intValue).toArray();
    }
}
// @lc code=end

